sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/implementation/websocket/WebSocketClientHandler.java

```diff
@@ -110,12 +121,61 @@ protected void channelRead0(ChannelHandlerContext ctx, Object msg) {
                 // close initiated from client, client already sent CloseWebSocketFrame
                 ch.close();
             }
+        } else if (msg instanceof TextWebSocketFrame || msg instanceof ContinuationWebSocketFrame) {
+            processFrame(msg);
+        } else {
+            loggerReference.get().atWarning().addKeyValue("frameType", msg).log("Unknown WebSocketFrame type.");
+        }
+    }
+
+    private void processFrame(WebSocketFrame frame) {
+        updateOpenTextFrame(frame.content());
+
+        if (!frame.isFinalFragment()) {
+            return;
+        }
+
+        final OpenTextFrame existing = UPDATER.getAndSet(this, null);
+
+        if (existing == null) {
+            final WebPubSubMessage message = messageDecoder.decode(frame.content().toString());
+            messageHandler.accept(message);
+
+            return;
+        }
+
+        existing.close();
+
+        final BinaryData collectedFrame = existing.getCollectedData();
+
+        if (collectedFrame != null) {
+            final String collected = collectedFrame.toString();
+
+            try {
+                final WebPubSubMessage deserialized = messageDecoder.decode(collected);
+                messageHandler.accept(deserialized);
+            } finally {
+                existing.dispose();
+            }
+        }
+    }
+
+    private void updateOpenTextFrame(ByteBuf content) {
+        content.retain();
+
+        final OpenTextFrame frame = new OpenTextFrame();
+        frame.add(content);
+
+        if (!UPDATER.compareAndSet(this, null, frame)) {
+            UPDATER.getAndUpdate(this, existing -> {
+                existing.add(content);
+                return existing;
+            });
         }
     }
 
     @Override
     public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
-        cause.printStackTrace();
         if (handshakeFuture != null && !handshakeFuture.isDone()) {
             handshakeFuture.setFailure(cause);
         }
```

Comment:
Any allocated resources (e.g. buffers) in the handler should be released on error, for example, if we were to use `CompositeByteBuf` (`this.cbb`) then it should be released `this.cbb.release()` here.

Reply:
regarding resource release, to be safe, we can additionally override the `handlerRemoved` and release.